sdk/src/ed25519_instruction.rs

```diff
@@ -379,4 +388,43 @@ pub mod test {
         );
         assert!(tx.verify_precompiles(&feature_set).is_err());
     }
+
+    #[test]
+    fn test_pairless_ed25519() {
+        solana_logger::setup();
+
+        let privkey = ed25519_dalek::Keypair::generate(&mut thread_rng());
+        let message_arr = b"hello";
+        let sig = privkey.sign(message_arr);
+        let mut instruction =
+            pairless_ed25519_instruction(privkey.public.to_bytes(), sig.to_bytes(), message_arr);
+        let mint_keypair = Keypair::new();
+        let feature_set = FeatureSet::all_enabled();
+
+        let tx = Transaction::new_signed_with_payer(
+            &[instruction.clone()],
+            Some(&mint_keypair.pubkey()),
+            &[&mint_keypair],
+            Hash::default(),
+        );
+
+        assert!(tx.verify_precompiles(&feature_set).is_ok());
+
+        let index = loop {
+            let index = thread_rng().gen_range(0, instruction.data.len());
+            // byte 1 is not used, so this would not cause the verify to fail
+            if index != 1 {
+                break index;
+            }
+        };
+
+        instruction.data[index] = instruction.data[index].wrapping_add(12);
+        let tx = Transaction::new_signed_with_payer(
+            &[instruction],
+            Some(&mint_keypair.pubkey()),
+            &[&mint_keypair],
+            Hash::default(),
+        );
+        assert!(tx.verify_precompiles(&feature_set).is_err());
+    }
```

Comment:
nit: do you mind refactoring this weird little part into a helper to also be used by `test_ed25519` ?